### PR TITLE
more tests for shl

### PR DESCRIPTION
**Pull request by mikelodder7**
Also fixes issue when shift % BIT_SIZE == 0.

### CODE REVIEW

File: crypto-bigint/src/uint/shl.rs [hunk 1]
```diff
@@ -21,17 +22,20 @@ impl<const LIMBS: usize> UInt<LIMBS> {
         }
 
         let shift_num = n / BIT_SIZE;
-        let shift_rem = n % BIT_SIZE;
+        let rem = n % BIT_SIZE;
+        let nz = is_nonzero(rem as Inner);
+        let lshift_rem = rem as Inner;
+        let rshift_rem = ct_select(0, (BIT_SIZE - rem) as Inner, nz);
```
Comment (tarcieri) [inline crypto-bigint/src/uint/shl.rs:28]: It seems like you could use `subtle` for this:

```suggestion
        let lshift_rem = rem as Inner;
        let rshift_rem = Inner::conditional_select(0, (BIT_SIZE - n) as Inner), !rem.ct_eq(&0));
```

Comment (mikelodder7) [reply crypto-bigint/src/uint/shl.rs:28]: Subtle is not a const fn

Comment (mikelodder7) [reply crypto-bigint/src/uint/shl.rs:28]: In order to have const fn for shift you have to use these. I would prefer subtle than reinventing the wheel

Comment (mikelodder7) [reply crypto-bigint/src/uint/shl.rs:28]: Once traits support const fn in stable then we can remove these

Comment (tarcieri) [reply crypto-bigint/src/uint/shl.rs:28]: Aha, makes sense

**Pull request merged**